modules/apps/site/site-cms-site-initializer/src/main/resources/META-INF/resources/js/main/components/forms/PermissionsTable.tsx

```diff
@@ -0,0 +1,387 @@
+) => {
+	return permissions.map(({actionIds, roleName}) => {
+		const actionIdsNotEnabled = actionIdsToEnable.filter(
+			(id) => !actionIds.includes(id)
+		);
+
+		return {
+			actionIds: roleNamesToEnable.includes(roleName)
+				? [...actionIds, ...actionIdsNotEnabled]
+				: actionIds.filter((item) => !actionIdsToEnable.includes(item)),
+			roleName,
+		};
+	});
+};
+
+function PermissionsTable({
+
+	/*
+	 * Actions to be listed as the row/column entries.
+	 */
+	actions = [
+		{
+			id: 'DELETE',
+			label: Liferay.Language.get('delete'),
+		},
+		{id: 'PERMISSIONS', label: Liferay.Language.get('permissions')},
+		{id: 'UPDATE', label: Liferay.Language.get('update')},
+		{id: 'VIEW', label: Liferay.Language.get('view')},
+	],
+
+	/*
+	 * Classname to apply extra styles to the table.
+	 */
+	className,
+
+	/*
+	 * Default group role that is between Guest and Owner, featured
+	 * in the Viewable By options and should be included as one of the roles
+	 * in "roles" (example: Organization Member, Power User,
+	 * Site Member).
+	 */
+	defaultGroupRole = 'Site Member',
+
+	/*
+	 * Default state for permissions.
+	 */
+	defaultPermissions = [
+		{
+			actionIds: ['VIEW'],
+			roleName: 'Guest',
+		},
+		{
+			actionIds: ['VIEW'],
+			roleName: 'Site Member',
+		},
+	],
+
+	/*
+	 * Checkboxes to be disabled for the guest role.
+	 */
+	guestUnsupportedActionIds = ['UPDATE'],
+
+	/*
+	 * Callback for when permissions gets changed.
+	 */
+	onChange = () => {},
+
+	/*
+	 * Orientation of the table (default is horizontal).
+	 */
+	reverse = false,
+
+	/*
+	 * Roles to be listed as the row/column entries.
+	 */
+	roles = [
+		{id: 'Guest', label: Liferay.Language.get('guest')},
+		{
+			id: 'Site Member',
+			label: Liferay.Language.get('site-member'),
+		},
+	],
+	...otherProps
+}: {
+	actions: {id: string; label: string}[];
+	className: string;
+	defaultGroupRole: string;
+	defaultPermissions: PermissionItem[];
+	guestUnsupportedActionIds: string[];
+	onChange: Function;
+	reverse: boolean;
+	roles: {id: string; label: string}[];
+}) {
+	const initialPermissions = useMemo(
+		() =>
+			roles.map(({id}) => ({
+				actionIds:
+					defaultPermissions.find(({roleName}) => id === roleName)
+						?.actionIds || [],
+				roleName: id,
+			})),
+		[roles, defaultPermissions]
+	);
+	const supportedActionIds: string[] = useMemo(
+		() => actions.map(({id}) => id),
+		[actions]
+	);
+	const supportedRoleNames: string[] = useMemo(
+		() => roles.map(({id}) => id),
+		[roles]
+	);
+
+	const [permissions, setPermissions] =
+		useState<PermissionItem[]>(initialPermissions);
+	const [viewableBy, setViewableBy] = useState<string>(
+		initialPermissions.every(({actionIds}) => actionIds.includes('VIEW')) // Guest -> all permissions enabled
+			? 'Guest'
+			: initialPermissions.every(
+						({actionIds}) => !actionIds.includes('VIEW')
+				  ) // Owner -> no permissions enabled
+				? 'Owner'
+				: defaultGroupRole // Group -> group permissions enabled
```

Comment:
[nit] Can you re-order the comment? I feel it follows the logic more if written like (with "view" included if I understand correctly) `All "view" permissions enabled -> Guest`.

I feel it reads more like "If all the view permissions are enabled, then set this as Guest"

Reply:
I guess it could read more clearly also if written like `Guest = All "view" permissions enabled"`

Reply:
Ok! I was thinking maybe the first way since as you said, it follows that order of "if _, then _"... I'll update the comment!